### PR TITLE
feature/fbfw-60/scroll-buttons

### PR DESCRIPTION
**Pull request by jchoi166**
### Phase Scroll Button

This PR adds a simple scroll button to be used in the Pill Scroller component for the Phase UI build. It makes use of a PhaseArrowIcon to display the action of the button (Forward or Backwards)

fbfw-60-scroll-button

### CODE REVIEW

File: beta-src/src/enums/UIState.ts [hunk 1]
```diff
@@ -9,4 +9,8 @@ enum UIState {
   INACTIVE = "inactive",
 }
 
-export default UIState;
+export enum ScrollButtonState {
+  BACK = "left",
+  FORWARD = "right",
+  ENABLED = "enabled",
+}
```
Comment (nebevets) [inline beta-src/src/enums/UIState.ts:16]: make ScrollButtonState it's own file. revert the changes you did to UIState

File: beta-src/src/interfaces/PhaseScroll.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { ScrollButtonState } from "../enums/UIState";
+
+export interface scrollButtonProps {
+  onClick?: React.MouseEventHandler<HTMLButtonElement> | undefined;
+  direction: ScrollButtonState;
+  disabled?: boolean;
```
Comment (nebevets) [inline beta-src/src/interfaces/PhaseScroll.ts:6]: alpha order props...i think it's better to have the props inside the component so that when people look at the component they can see them in the same context.

File: beta-src/src/webDiplomacyTheme.ts [hunk 3]
```diff
@@ -86,6 +86,12 @@ const webDiplomacyTheme = createTheme({
         root: {
           borderRadius: 18,
           padding: "10px 18px",
+          "&.WDScrollButton, &.WDScrollButton:hover, &.WDScrollButton:focus": {
```
Comment (nebevets) [inline beta-src/src/webDiplomacyTheme.ts:89]: let's not add this to the theme...you can use sx prop on Button to override the styles instead.

File: beta-src/src/components/svgr-components/WDPhaseArrowIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import * as React from "react";
+import { scrollButtonProps } from "../../interfaces/PhaseScroll";
+import { ScrollButtonState } from "../../enums/UIState";
+
+const WDPhaseArrowIcon: React.FC<scrollButtonProps> = function ({
+  disabled = false,
```
Comment (nebevets) [inline beta-src/src/components/svgr-components/WDPhaseArrowIcon.tsx:6]: let's define the default props like they are in WDButton for consistency

File: beta-src/src/components/ui/WDScrollButton.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import { scrollButtonProps } from "../../interfaces/PhaseScroll";
+import WDPhaseArrowIcon from "../svgr-components/WDPhaseArrowIcon";
+
+const WDScrollButton: React.FC<scrollButtonProps> = function ({
+  onClick = undefined,
+  direction,
+  disabled = false,
+}): React.ReactElement {
+  return (
+    <Button
```
Comment (nebevets) [inline beta-src/src/components/ui/WDScrollButton.tsx:12]: can we use WDButton instead?

File: beta-src/src/components/ui/WDScrollButton.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      className="WDScrollButton"
```
Comment (nebevets) [inline beta-src/src/components/ui/WDScrollButton.tsx:13]: lets use sx instead of a className

**Review by nebevets — CHANGES_REQUESTED** [review]

**Review by bryancodazen — CHANGES_REQUESTED** [review]

File: beta-src/src/components/ui/WDScrollButton.tsx (hunk 5 above)
Comment (jchoi166) [reply beta-src/src/components/ui/WDScrollButton.tsx:12]: I could, but I would have to add a "direction" prop to WDButton so that it knows which arrow to generate for the scroll button. I didn't want to mess with WDButton so I decided to create my own, but I can make that change then its an easy swap!

File: beta-src/src/enums/UIState.ts [hunk 7]
```diff
@@ -9,4 +9,8 @@ enum UIState {
   INACTIVE = "inactive",
 }
 
-export default UIState;
+export enum ScrollButtonState {
+  BACK = "left",
+  FORWARD = "right",
+  ENABLED = "enabled",
```
[the remark this answers is not in the document]
Comment (jchoi166) [reply beta-src/src/enums/UIState.ts:15]: LOL that makes more sense. 

And I actually think the enabled enum isn't necessary. I think I put that there when I was building the phase icon, but it doesn't use this enum anymore so I'll go ahead and delete it!

File: beta-src/src/components/ui/WDScrollButton.tsx (hunk 5 above)
Comment (nebevets) [reply beta-src/src/components/ui/WDScrollButton.tsx:18]: if you did that, it would be styled already and we could keep them in sync. just make them optional and set default values...then when you use it here you can pass the disableRipple, etc.

File: beta-src/src/components/ui/WDScrollButton.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import * as React from "react";
+import Button from "@mui/material/Button";
+import ScrollButtonState from "../../enums/ScrollButton";
+import WDPhaseArrowIcon from "../svgr-components/WDPhaseArrowIcon";
+
+interface scrollButtonProps {
```
Comment (nebevets) [inline beta-src/src/components/ui/WDScrollButton.tsx:6]: interfaces are Capitalized

**Review by nebevets — CHANGES_REQUESTED** [review]
modify WDButton so it can work for your use case.

**Comment by jchoi166** [comment]
@bryancodazen Re-requesting review!

File: beta-src/src/components/svgr-components/WDPhaseArrowIcon.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  direction: ScrollButtonState;
+  disabled?: boolean;
+}
+
+const WDPhaseArrowIcon: React.FC<phaseArrowProps> = function ({
+  disabled,
+  direction,
+}): React.ReactElement {
+  return (
+    <svg fill="none" height={15} width={8} xmlns="http://www.w3.org/2000/svg">
```
Comment (nebevets) [inline beta-src/src/components/svgr-components/WDPhaseArrowIcon.tsx:14]: height and width can just be strings if we aren't changing them

**Review by nebevets — APPROVED** [review]
fix a couple of nits...otherwise LGTM

File: beta-src/src/components/svgr-components/WDPhaseArrowIcon.tsx (hunk 9 above)
Comment (jchoi166) [reply beta-src/src/components/svgr-components/WDPhaseArrowIcon.tsx:14]: Changed!

**Review by bryancodazen — CHANGES_REQUESTED** [review]

**Review by bryancodazen — APPROVED** [review]
